### DISCUSSION TITLE
Why not create a common `Rejection` trait?

### DISCUSSION THREAD

**Discussion by alekseysidorov** [tokio-rs/axum#2319] [Ideas]
Why not create a common `Rejection` trait?

At this moment, there are a lot of structs in the rejection module. All of them have `body_text` and `status` methods. Why not create a `Rejection` trait for all of them?

This might be helpful for generic rejection errors handling. Imagine something like following:

```rust
struct MyErrorType { ... };

// With an `Rejection` trait I be able to write something like this
impl<T: Rejection> from T for MyErrorType {
    //
}
```

**Comment by davidpdrsn** [comment]
Do you have an actual use case for such a trait or is it just speculation? I'm not a fan of abstracting things more than necessary. You could also define such a trait yourself if you needed it in your own code base.
